docs/SPIRVRepresentationInLLVM.rst

```diff
@@ -239,3 +297,55 @@ For example:
   !9 = !{!7, i32 32}     ; independent forward progress is required for 'kernel2'
   !10 = !{i16 6, i16 123} ; 6 - Generator Id, 123 - Generator Version 
 
+Additional requirements for LLVM module
+=======================================
+
+Target triple and datalayout string
+-----------------------------------
+
+Target triple architecture must be ``spir`` (32-bit architecture) or ``spir64``
+(64-bit architecture) and ``datalayout`` string must be aligned with OpenCL
+environment specification requirements for data type sizes and alignments (e.g.
+3-element vector must have 4-element vector alignment). For example:
+
+.. code-block:: llvm
+   target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+   target triple = "spir-unknown-unknown"
+
+Target triple architecture is translated to
+`addressing model operand <https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_addressing_model_a_addressing_model>`_
+of
+`OpMemoryModel <https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#_a_id_mode_setting_a_mode_setting_instructions>`_
+SPIR-V instruction.
+
+- ``spir`` -> Physical32
+- ``spir64`` -> Physical64
+
+Calling convention
+------------------
+
+OpEntryPoint information is represented in LLVM IR in calling convention.
+A function with `spir_kernel` calling convention will be translated as an entry
+point to SPIR-V module.
+
+Function metadata
+-----------------
+
+Some kernel parameter information is stored in LLVM IR as a function metadata.
+
+For example:
+
+- !kernel_arg_addr_space !1
+- !kernel_arg_access_qual !2
+- !kernel_arg_type !3
+- !kernel_arg_base_type !4
+- !kernel_arg_type_qual !5
+
+**TODO**: Are all these metadata optional? Access qualifiers are translated for
```

Comment:
Summoning @StuartDBrady for a discussion.

Reply:
> Are all these metadata optional?

I think so.  Clang started generating this metadata as of commit `fb36ede52ebb ("Generate metadata to implement the -cl-kernel-arg-info option. OpenCL 1.2 spec. 5.7.3.", 2013-03-24)`, with the purpose of supporting `clGetKernelArgInfo`.  The OpenCL API spec says that 

> Kernel argument information is only available if the program object associated with kernel is created with clCreateProgramWithSource and the program executable was built with the -cl-kernel-arg-info option specified in options argument to clBuildProgram or clCompileProgram.

So an implementation should not need to support kernel argument info in the SPIR-V case.

Reply:
Changed TODO to NOTE.